packages/jaeger-ui/src/components/TracePage/TraceGraph/calculateTraceDagEV.tsx

```diff
@@ -81,7 +81,12 @@ export function calculateTraceDag(trace: Trace): TraceDag<TSumSpan & TDenseSpanM
   const dag = new TraceDag<TSumSpan & TDenseSpanMembers>();
 
   baseDag.nodesMap.forEach(node => {
-    const ntime = node.members.reduce((p, m) => p + m.span.duration, 0);
+    const ntime = node.members.reduce(
+      (mdr, m) =>
+        mdr.add(m.span.startTime, m.span.startTime + (m.span.duration <= 0 ? 0 : m.span.duration - 1)),
+      new DRange()
+    ).length;
+    const totalTime = node.members.reduce((p, m) => p + m.span.duration, 0);
```

Comment:
@GLVSKiriti this sounds similar to the other fix you made in another view. Is there a way to consolidate these calculations?

Reply:
I don't know exactly but I will take a look on this!